### DISCUSSION TITLE
[RFC] Why do we import from Notion via the internet?

### DISCUSSION THREAD

**Discussion by angeloreale** [anyproto/contributors#227] [Anytype apps]
Anytype is meant to be local-first.
Notion can export its entire space as csv
Why does Anytype needs an integration to import from it?

Solution:
- Let's work on an all encompassing csv importer that understands Notion zip files too.

**Comment by angeloreale** [comment]
Moreover, if it can use my token, why not delegate the import to the cloud / do it in in chunks (storing progress locally, as we disconnect and reconnect)?

**Reply by angeloreale** [reply]
Also, why not implement a consensus strategy for reconciliation? ORT? Paxos? AnyPaxos?

**Reply by angeloreale** [reply]
Can we use Anysync?

**Reply by angeloreale** [reply]
Realm.io uses Raft, I like it. It can help on the mobile app too.

**Reply by angeloreale** [reply]
I think they call it App Services these days.

**Comment by AnastasiaShemyakinskaya** [comment]
> Anytype is meant to be local-first. Notion can export its entire space as csv Why does Anytype needs an integration to import from it?
> 
> Solution:
> 
> * Let's work on an all encompassing csv importer that understands Notion zip files too.

Import from Notion via api is just another way to import your data from Notion. If it doesn't work for you, we have Markdown import, which can be used for importing backup from Notion.

**Comment by th0rgall** [comment]
> Let's work on an all encompassing csv importer that understands Notion zip files too.

Did you check whether the .csv export of Notion has the same data quality as the output from its REST API?

Last time I checked (some years ago), this was not the case. Even internal links between Notion pages were inconsistent. I can imagine that the REST API delivers more complete metadata that is helpful when trying to clone the content as accurately as possible.
